osbs/utils/__init__.py

```diff
@@ -403,17 +420,16 @@ def parse(cls, image_name):
 
         if len(s) == 2:
             if '.' in s[0] or ':' in s[0]:
-                result.registry = s[0]
+                result = result.replace(registry=s[0], repo=s[1])
             else:
-                result.namespace = s[0]
+                result = result.replace(namespace=s[0], repo=s[1])
         elif len(s) == 3:
-            result.registry = s[0]
-            result.namespace = s[1]
-        result.repo = s[-1]
+            result = result.replace(registry=s[0], namespace=s[1], repo=s[2])
 
         for sep in '@:':
             try:
-                result.repo, result.tag = result.repo.rsplit(sep, 1)
+                repo, tag = result.repo.rsplit(sep, 1)
+                result = result.replace(repo=repo, tag=tag)
```

Comment:
Instead of `replace`-ing the previous result, could we just figure out the `registry`, `repo` etc. variables and create the ImageName object at the end?